### PR TITLE
[WFLY-11503] move getClassLoader action into doPrivileged block.

### PR DESCRIPTION
**Pull request by soul2zimate**
https://issues.jboss.org/browse/WFLY-11503

### CODE REVIEW

**Review by rhusar — CHANGES_REQUESTED** [review]
This doesn't seem right to me, rather it breaks the test so it won't discover the actual issues. The issue seems to be `org.jboss.as.weld.injection.InjectionTargets#createInjectionTarget` line 68.

**Review by manovotn — APPROVED** [review]

**Review by rhusar — APPROVED** [review]
Looks good to me now. (PS: I have just noticed many PR like you had your original ones which just workaround the SM issues, rather than fix them, but they do get merged... so that's how I stumbled upon this)

File: weld/subsystem/src/main/java/org/jboss/as/weld/injection/InjectionTargets.java [hunk 1]
```diff
@@ -65,7 +69,13 @@ private InjectionTargets() {
              * If this happens, use a combination of a bean archive identifier and class' classloader hashCode as the BDA ID.
              * This breaks AnnotatedType serialization but that does not matter as these are non-contextual components.
              */
-            final String bdaId = beanManager.getId() + componentClass.getClassLoader().hashCode();
+            final ClassLoader classLoader;
+            if (WildFlySecurityManager.isChecking()) {
+                classLoader = AccessController.doPrivileged((PrivilegedAction<ClassLoader>) () -> componentClass.getClassLoader());
```
Comment (kabir) [inline weld/subsystem/src/main/java/org/jboss/as/weld/injection/InjectionTargets.java]: @soul2zimate We should not use lamdas generally. Also there is a GetClassLoaderAction which can be used to do the work for the privileged case

Comment (soul2zimate) [reply weld/subsystem/src/main/java/org/jboss/as/weld/injection/InjectionTargets.java]: I have updated this to use GetClassLoaderAction.

**Pull request merged**